### PR TITLE
Forward tags to registered ModelVersion in Model.log()

### LINKED ISSUE #24101: [BUG] tags passed to log_model are not propagated to the registered ModelVersion when registered_model_name is specified
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- Client: 3.14.0
- Tracking server: 3.14.0

### System information

- **OS Platform and Distribution**: Ubuntu 24.04
- **Python version**: 3.11

### Describe the problem

When calling `mlflow.<flavor>.log_model(..., tags={"key": "value"}, registered_model_name="my-model")`, the `tags` argument is silently dropped and never attached to the resulting `ModelVersion`. The tags end up only on the `LoggedModel` entity, which is invisible when inspecting the model registry.

### Root cause

In `mlflow.models.model.Model.log` (`mlflow/models/model.py`), after the `LoggedModel` is created with the user-supplied tags, `_register_model` is called **without** forwarding `tags`:

```python
# model.py – Model.log()
if registered_model_name is not None:
    registered_model = mlflow.tracking._model_registry.fluent._register_model(
        f"models:/{model.model_id}",
        registered_model_name,
        await_registration_for=await_registration_for,
        local_model_path=local_path,
        # tags is NOT forwarded here ← bug
    )
```

`_register_model` already accepts a `tags` keyword argument and forwards it correctly to `client._create_model_version`. The fix is a one-liner: add `tags=tags` to the call above.

### Expected behaviour

Tags passed to `log_model` should be attached to both the `LoggedModel` entity and the registered `ModelVersion` (when `registered_model_name` is provided), consistent with what a caller would intuitively expect.

### Proposed fix

```python
# model.py – Model.log()
if registered_model_name is not None:
    registered_model = mlflow.tracking._model_registry.fluent._register_model(
        f"models:/{model.model_id}",
        registered_model_name,
        await_registration_for=await_registration_for,
        local_model_path=local_path,
        tags=tags,  # ← forward tags to ModelVersion
    )
````

### Tracking information

_No response_

### Code to reproduce issue

```python
import mlflow
import mlflow.sklearn
from sklearn.linear_model import LinearRegression

mlflow.set_tracking_uri("http://localhost:5000")

model = LinearRegression().fit([[1], [2]], [1, 2])

with mlflow.start_run():
    info = mlflow.sklearn.log_model(
        model,
        name="my-model",
        registered_model_name="my-model",
        tags={"stage": "training", "framework": "sklearn"},
    )

client = mlflow.MlflowClient()

# Tags ARE on the LoggedModel
print(client.get_logged_model(info.model_id).tags)
# → {'stage': 'training', 'framework': 'sklearn'}

# Tags are NOT on the ModelVersion
mv = client.get_model_version("my-model", info.registered_model_version)
print(mv.tags)
# → {}  ← expected {'stage': 'training', 'framework': 'sklearn'}

```

### Stack trace

No errors on stack trace.

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

### PR DESCRIPTION
**Pull request by Ankita7033**
### Related Issues/PRs
Closes #24101

### What changes are proposed in this pull request?
When calling `log_model(..., tags={...}, registered_model_name=...)`, the tags were dropped from the registered ModelVersion. `Model.log()` called `_register_model(...)` without forwarding the `tags` argument, so the tags only reached the LoggedModel and never propagated to the ModelVersion. This forwards `tags=tags` to the `_register_model(...)` call in `Model.log()`.

### How is this PR tested?
- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

Added a regression test in `tests/models/test_model.py` (`test_model_log_tags_propagated_to_registered_model_version`) that logs a model with both `tags` and `registered_model_name`, then asserts the resulting ModelVersion carries the tags. Verified passing locally along with the existing model-version and model-log tests.

### Does this PR require documentation update?
- [x] No.

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?
- [x] No.

#### Is this a user-facing change?
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fixed a bug where tags passed to `log_model(..., registered_model_name=...)` were not propagated to the registered model version.

#### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors

<a name="release-note-category"></a>
#### How should the PR be classified in the release notes? Choose one:
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes

#### Is this PR a critical bugfix or security fix that should go into the next patch release?
<details>
<summary>What is a minor/patch release?</summary>
- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Minor releases are expected to contain larger changes, such as new features and improvements. Non-critical bug fixes and doc updates can be included as well. By default, your PR should target the next minor release.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Patch releases are typically only performed when there has been a major regression or bug in the latest release. For the sake of stability, your PR should not be included in a patch release unless it is a critical fix, or if the risk level of your PR is exceedingly low.
</details>
- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

**Comment by PattaraS** [comment]
Pushed a small lint cleanup directly to your branch (@Ankita7033, hope that's alright): removed a stray trailing-whitespace line at the `_log_v2` register call site and applied formatter spacing to the new test.

Good call not adding `tags=tags` at the `_log_v2` site — it doesn't take a `tags` param and its `**kwargs` aren't threaded to `_register_model`, so it'd raise `NameError`. The Spark/`mlflowdbfs` path not forwarding tags to registration is a separate, pre-existing gap, out of scope here. The `Model.log` fix is the one #24101 is about, and it's correct. LGTM once CI is green.

**Review by PattaraS — APPROVED** [review]
